src/hb-ot-layout-common.hh

```diff
@@ -193,6 +193,71 @@ struct Record
   DEFINE_SIZE_STATIC (6);
 };
 
+struct Script;
+struct LangSys;
+struct Feature;
+
+struct LangSysRecord : Record<LangSys>
+{
+  bool subset (hb_subset_context_t    *c,
+	       const hb_map_t         *feature_index_map,
+	       const void             *src_base,
+	       const void             *dst_base) const
+  {
+    TRACE_SERIALIZE (this);
+    auto *out = c->serializer->embed (this);
+    if (unlikely (!out)) return_trace (false);
+    bool ret = out->offset.serialize_subset (c, offset, src_base, dst_base, feature_index_map);
+    return_trace (ret);
+  }
+};
+
+struct ScriptRecord : Record<Script>
+{
+  bool subset (hb_subset_context_t         *c,
+	       RecordList_subset_context_t *record_list_context,
```

Comment:
Can you create a hb_subset_layout_context_t which holds, hb_subset_context_t, max langsys and script, the feature and lookup index maps, and table tag. Then pass that around instead of the individual items.